subprojects/docs/src/snippets/dependencyManagement/customizingResolution-capabilitySubstitutionRule/groovy/build.gradle

```diff
@@ -20,8 +20,8 @@ configurations.testCompileClasspath {
 // end::substitution_rule[]
 
 tasks.register('resolve') {
-    inputs.files(configurations.testCompileClasspath)
```

Comment:
We don't want to remove the declaration of this task's inputs.  Without the input being declared, the dependency between the `resolve` and the Java compilation tasks won't be identified by Gradle.

Other than that, your changes look good.

Reply:
@abstratt Sorry about this, I put back the line as it was before.